File: app/src/main/java/com/github/galcyurio/freetodo/di/module/DatabaseModule.java
package com.github.galcyurio.freetodo.di.module;

import com.github.galcyurio.freetodo.data.source.DbHelper;

import javax.inject.Singleton;

import android.content.Context;
import dagger.Module;
import dagger.Provides;

/**
 * @author galcyurio
 */
@Module
public class DatabaseModule {

    @Singleton
    @Provides
    DbHelper provideDbHelper(Context context) {
        return new DbHelper(context);
    }

}

File: app/src/main/java/com/github/galcyurio/todo/DatabaseModule.kt
package com.github.galcyurio.todo

import android.content.Context
import androidx.room.Room
import com.github.galcyurio.todo.data.local.AppDatabase
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.qualifiers.ApplicationContext
import dagger.hilt.components.SingletonComponent
import javax.inject.Singleton

@InstallIn(SingletonComponent::class)
@Module
interface DatabaseModule {
    companion object {
        @Provides
        @Singleton
        fun provideAppDatabase(@ApplicationContext context: Context): AppDatabase {
            return Room.databaseBuilder(
                context,
                AppDatabase::class.java,
                "tasks.db"
            ).build()
        }
    }
}
